src/Functions/checkPartMetadataCache.cpp

```diff
@@ -0,0 +1,156 @@
+#include "config_core.h"
+
+#if USE_ROCKSDB
+#include <Functions/FunctionFactory.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypeFixedString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Storages/MergeTree/MergeTreeData.h>
+#include <Storages/MergeTree/IMergeTreeDataPart.h>
+#include <Columns/ColumnArray.h>
+#include <Columns/ColumnTuple.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnFixedString.h>
+#include <Columns/ColumnsNumber.h>
+#include <Interpreters/Context.h>
+#include <Common/hex.h>
+#include <Core/Field.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int ILLEGAL_COLUMN;
+}
+
+class FunctionCheckPartMetadataCache : public IFunction, WithContext
+{
+public:
+    using uint128 = IMergeTreeDataPart::uint128;
+    using DataPartPtr = MergeTreeData::DataPartPtr;
+    using DataPartState = MergeTreeData::DataPartState;
+    using DataPartStates = MergeTreeData::DataPartStates;
+
+
+    static constexpr auto name = "checkPartMetadataCache";
```

Comment:
By the way, does generic function a good way to implement this, looks like some system command will be better (or even built this check into `CHECK TABLE`)

Reply:
I think `CHECK TABLE` query is better. I will implement checking in this way later.

Reply:
I implement the logic of checking consistency of metadata in `CHECK TABLE`.  
See StorageMergeTree::checkData and StorageMergeTree::checkData and StorageReplicatedMergeTree::checkData